lib/circuitry/processors/threader.rb

```diff
@@ -0,0 +1,22 @@
+require 'circuitry/processor'
+
+module Circuitry
+  module Processors
+    module Threader
+      class << self
+        include Processor
+
+        def process(&block)
+          raise ArgumentError, 'no block given' unless block_given?
+          pool << Thread.new { safely_process(&block) }
```

Comment:
It's unbounded, so it's up to the implementor to flush it.  The solution you proposed would only work for thread pooling, not batching.  I wonder if there's a better way to expose the pooling while allowing the user to implement a custom mechanism.  Is it worth worrying about at this point?